lib/statsd/instrument/udp_connection.rb

```diff
@@ -30,6 +31,7 @@ def type
       def socket
         @socket ||= begin
           socket = UDPSocket.new
+          socket.setsockopt(Socket::SOL_SOCKET, Socket::SO_SNDBUF, @max_packet_size)
```

Comment:
Like you mentioned in the PR description, there's a chance that the user could set a value higher than what the operating system would support. If that happens, wouldn't we still be liable to unexpectedly lose data in the same way the PR describes?

I think it _might_ be worth validating the `@max_packet_size` somehow (probably before we get down to this point, but 🤷 wherever it makes the most sense). I'm not sure if it's best to manually (or programmatically, somehow) check what the actual OS limits are and validate configuration accordingly, or possibly just check and see if we're able to set whatever the user asked for (and then take action accordingly: log a big-ol' warning, lower the max packet size to whatever the buffer is actually set to at this point, run around with our hair on fire, etc).

It's pretty easy to ask the socket what that value is before/after you try and modify it:

```ruby
irb(main):001> require 'socket'
=> true
irb(main):002> s = UDPSocket.new
=> #<UDPSocket:fd 6, AF_INET, 0.0.0.0, 0>
irb(main):003> s.getsockopt(Socket::SOL_SOCKET, Socket::SO_SNDBUF).int
=> 9216
irb(main):004> s.setsockopt(Socket::SOL_SOCKET, Socket::SO_SNDBUF, 99_999_999)
=> 0
irb(main):005> s.getsockopt(Socket::SOL_SOCKET, Socket::SO_SNDBUF).int
=> 8388608
irb(main):006>
```

Of course, IDK if this works the same way on Windows, or if that matters much to us.

Reply:
That is an awesome call out Andrew! ❤️ Will change it to do so.